### PR TITLE
add integration testing

### PR DESCRIPTION
**Pull request by JPadovano1483**
Adds Kind-based end-to-end integration tests for the management stack and backbone site bootstrap. Tests run via Vitest against a local Kind cluster with PostgreSQL, cert-manager, Keycloak, management-controller, Skupper (multi-van), and site-controller. Test coverage is outlined in the README in /tests/integration/kind.

Testing:
- run `pnpm install`
- run `pnpm test:integration:local`  to spin up a Kind cluster, run integration tests, and tear down the cluster; OR
- manually bring up the cluster and run tests with `pnpm cluster:up`, `pnpm test:integration`, and `pnpm cluster:down` to tear down the cluster

Depends on #161 
Part of #126

### CODE REVIEW

**Review by ted-ross — APPROVED** [review]
There is a typo in the second bullet in the PR description.

This PR looks very good.  I would like Fernando to also provide a review.

**Comment by JPadovano1483** [comment]
> There is a typo in the second bullet in the PR description.
> 
> This PR looks very good. I would like Fernando to also provide a review.

Thanks! I will wait for @fgiorgetti to review before merging.

One question. In the skupper-install script, I used the multi-van router image from the tedlross quay repo. It looks like we swiched to using the main image from the skupper repo? I also pulled all the CRDs from Fernando's repo (I had a recent CRD regression that caused all sorts of issues, so I may have overcompensated by pulling all of them). If either or you wouldn't mind taking a look at those and letting me know what the correct images and CRDs are, that would be great.

It may be worth discussing having a more centralized location for these. The getting-started doc is currently out of sync, so there doesn't seem to really be a great source of truth for what we need to be running.

File: tests/integration/helpers/bootstrap.js [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+  }
+}
+
+/**
+ * Wait until a label selector has at least one Running pod.
+ * @param {string} labelSelector
+ * @param {string} [namespace]
+ * @param {number} [timeoutMs]
+ */
+export async function waitForRunningPod(
```
Comment (fgiorgetti) [inline tests/integration/helpers/bootstrap.js]: Maybe move this to `kubectl.js`

File: tests/integration/helpers/bootstrap.js [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+    parts.push(`init=${init.state.waiting.reason}`)
+  }
+  if (init?.state?.terminated?.reason) {
+    parts.push(
+      `init=${init.state.terminated.reason} exit=${init.state.terminated.exitCode}`,
+    )
+  }
+  try {
+    const logs = kubectl(
+      ["logs", pod.metadata.name, "-c", "config-init", "--tail=20"],
```
Comment (fgiorgetti) [inline tests/integration/helpers/bootstrap.js]: should it be static to `config-init`?

File: tests/integration/kind/scripts/install-skupper.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+# Install Skupper v2 controller with multi-van CRDs and images.
+set -euo pipefail
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+# shellcheck source=../config.sh
+source "${ROOT}/config.sh"
+
+echo "Ensuring namespace ${SITE_NAMESPACE}..."
```
Comment (fgiorgetti) [inline tests/integration/kind/scripts/install-skupper.sh]: What is the purpose of ensuring SITE_NAMESPACE ?

File: tests/integration/kind/scripts/install-skupper.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  }}
+]'; then
+  echo "ClusterRole patch skipped (rules may already exist)"
+fi
+
+echo "Waiting for skupper-controller..."
+kubectl --context "${KUBECTL_CONTEXT}" -n "${SKUPPER_NAMESPACE}" rollout status deployment/skupper-controller --timeout=300s
+
+echo "Removing router deployments created with stock kube-adaptor (controller will recreate them)..."
+for ns in "${SITE_NAMESPACE}" "${NAMESPACE}"; do
```
Comment (fgiorgetti) [inline tests/integration/kind/scripts/install-skupper.sh]: Is this clean up really needed here?

**Review by fgiorgetti — COMMENTED** [review]
A few minor comments only, but overall it seems to work well.

I noticed that once I start the kind cluster and immeditally run the integration tests, it is failing 1 test only once, but if I run it again, it passes:

 ❯  integration  tests/integration/kind/specs/backbone-bootstrap.test.js (8 tests | 1 failed) 10203ms
     ✓ Postgres has backbone site seeded and bootstrapped  414ms
     **× site-controller deployment is available in site-a 57ms**
     ✓ skupper-router pod is running in site-a 49ms
     ✓ site-controller startup log markers are present 114ms
       ✓ GET /api/v1alpha1/hostnames returns manage access point ingress bundle  3031ms
     ✓ manage access point ingress appears in Postgres after state sync  906ms
     ✓ manage access-point TLS secret exists in site namespace 112ms
     ✓ management-controller logs AMQP connection to site manage access point 163ms
 ✓  integration  tests/integration/kind/specs/mgmt-health.test.js (7 tests) 3956ms
     ✓ Postgres schema tables exist  2260ms
     ✓ Configuration seed row is present  489ms
     ✓ ManagementControllers row created on startup  521ms
 ✓  integration  tests/integration/kind/specs/mgmt-auth-api.test.js (6 tests) 1597ms
     ✓ POST /backbones creates a backbone and DELETE removes it  601ms
 ✓  integration  tests/integration/kind/specs/mgmt-certs.test.js (5 tests) 632ms

Another comment that is not appropriate for this PR, but something for an upcoming enhancement just to keep in mind (like when we enable integration tests to run through the CI), would be to add a test reporter, to write results in JSON/JUnit format. As those resources could be stored so we can compare results against separate runs.

**Comment by JPadovano1483** [comment]
> A few minor comments only, but overall it seems to work well.
> 
> I noticed that once I start the kind cluster and immeditally run the integration tests, it is failing 1 test only once, but if I run it again, it passes:
> 
> ❯ integration tests/integration/kind/specs/backbone-bootstrap.test.js (8 tests | 1 failed) 10203ms ✓ Postgres has backbone site seeded and bootstrapped 414ms **× site-controller deployment is available in site-a 57ms** ✓ skupper-router pod is running in site-a 49ms ✓ site-controller startup log markers are present 114ms ✓ GET /api/v1alpha1/hostnames returns manage access point ingress bundle 3031ms ✓ manage access point ingress appears in Postgres after state sync 906ms ✓ manage access-point TLS secret exists in site namespace 112ms ✓ management-controller logs AMQP connection to site manage access point 163ms ✓ integration tests/integration/kind/specs/mgmt-health.test.js (7 tests) 3956ms ✓ Postgres schema tables exist 2260ms ✓ Configuration seed row is present 489ms ✓ ManagementControllers row created on startup 521ms ✓ integration tests/integration/kind/specs/mgmt-auth-api.test.js (6 tests) 1597ms ✓ POST /backbones creates a backbone and DELETE removes it 601ms ✓ integration tests/integration/kind/specs/mgmt-certs.test.js (5 tests) 632ms
> 
> Another comment that is not appropriate for this PR, but something for an upcoming enhancement just to keep in mind (like when we enable integration tests to run through the CI), would be to add a test reporter, to write results in JSON/JUnit format. As those resources could be stored so we can compare results against separate runs.

Hmm, I think I see why this might be happening, though I haven't been able to replicate the issue. 

This is a good suggestion. I'll make note of it in #157

File: tests/integration/kind/scripts/install-skupper.sh (hunk 4 above)
Comment (JPadovano1483) [reply tests/integration/kind/scripts/install-skupper.sh]: No this can be removed. I think the only situation where it would do anything would be if you were manually playing around with the cluster and deployed a router with different images. Rerunning cluster:up would ensure they all had the same images afterwards. Probably not a likely edge case though...

File: tests/integration/kind/scripts/install-skupper.sh (hunk 3 above)
Comment (JPadovano1483) [reply tests/integration/kind/scripts/install-skupper.sh]: Only necessary here in conjunction with the router delete below. This can be moved to the bootstrapping function if the router deletion is removed.

File: tests/integration/helpers/bootstrap.js (hunk 2 above)
Comment (JPadovano1483) [reply tests/integration/helpers/bootstrap.js]: This can probably made more general, especially if moved to kube.js. Would the suggestion here be to find any init containers that didn't complete and grab logs from those, rather than hardcoding the config-init container?

Comment (fgiorgetti) [reply tests/integration/helpers/bootstrap.js]: Correct. Get the list of initContainer names and eventually iterate through them.

File: tests/integration/kind/scripts/install-skupper.sh (hunk 3 above)
Comment (fgiorgetti) [reply tests/integration/kind/scripts/install-skupper.sh]: Alright, I will leave it up to you.

**Review by fgiorgetti — APPROVED** [review]
LGTM! Worked great now.

**Comment by fgiorgetti** [comment]
> One question. In the skupper-install script, I used the multi-van router image from the tedlross quay repo. It looks like we swiched to using the main image from the skupper repo?

Correct. You should use the router image from the skupper repo now.

> I also pulled all the CRDs from Fernando's repo (I had a recent CRD regression that caused all sorts of issues, so I may have overcompensated by pulling all of them). If either or you wouldn't mind taking a look at those and letting me know what the correct images and CRDs are, that would be great.
> It may be worth discussing having a more centralized location for these. The getting-started doc is currently out of sync, so there doesn't seem to really be a great source of truth for what we need to be running.

Right now, the CRDs have to be consumed from the location where you have it in the install skupper script (https://github.com/fgiorgetti/skupper/tree/refs/heads/multi-van/config/crd/bases).

My goal is to get these updated CRDs reviewed by the whole group and merged into the main branch of the skupper repo, so we can rely on the official source of truth.
